src/op_handlers/far_call.rs

```diff
@@ -262,7 +262,6 @@ pub fn far_call(
 
             // set calldata pointer
             vm.set_register(1, TaggedValue::new_pointer(forward_memory.encode()));
-            vm.current_context_mut()?.caller = address_from_u256(&vm.get_register(15).value);
```

Comment:
I don't exactly understand this bit. Is this due to static mode being in effect?

Reply:
No, in mimic calls the caller is set to the value of register 15

On the formal spec:
```
Mimic call sets:
this_address <- destination address;
msg_sender <- value of r15;
context <- value of context register gs_context_u128
```